lib/SubInput.js

```diff
@@ -0,0 +1,55 @@
+/*
+  Copyright © 2018 Andrew Powell
+
+  This Source Code Form is subject to the terms of the Mozilla Public
+  License, v. 2.0. If a copy of the MPL was not distributed with this
+  file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+  The above copyright notice and this permission notice shall be
+  included in all copies or substantial portions of this Source Code Form.
+*/
+
+// A PostCSS Input that exposes a substring of a larger Input as though it were
+// the entire text to be parsed.
+module.exports = class SubInput {
```

Comment:
Without seeing how this is used (via tests) I can't comment on whether or not this is a good name for this. 

Also, why does this not inherit from `Input`?